commcare_connect/opportunity/tasks.py

```diff
@@ -520,6 +524,73 @@ def get_learning_user_credentials(credential_config):
     )
 
 
+@celery_app.task()
+def issue_credentials_to_users():
+    """
+    Issue user credentials to PersonalID for all users who have not been issued credentials.
+    """
+
+    def parse_credential_payload_item(users_credential):
+        return {
+            "usernames": users_credential["usernames"],
+            "title": UserCredential.get_title(
+                credential_type=users_credential["credential_type"],
+                level=users_credential["level"],
+                delivery_type_name=users_credential["delivery_type__name"],
+            ),
+            "type": users_credential["credential_type"],
+            "level": users_credential["level"],
+            "slug": users_credential["opportunity__id"],
+            "opportunity_id": users_credential["opportunity__id"],
+        }
+
+    unissued_credentials_qs = get_unissued_user_credentials_queryset()
+    payload_size_counter = 0
+    credentials_payload_items = []
+    index_to_credential_ids_set_mapper = {}
+
+    for index, users_credential in enumerate(unissued_credentials_qs.iterator(chunk_size=MAX_CREDENTIALS_PER_REQUEST)):
```

Comment:
In the interest of conserving memory I thought it's a good idea to only load the number of records we know we're going to end up sending to PersonalID at a time from the DB.